src/main/java/uk/gov/companieshouse/api/accounts/service/impl/AbstractServiceImpl.java

```diff
@@ -23,12 +27,12 @@ public AbstractServiceImpl(MongoRepository mongoRepository,
     }
 
     @Override
-    public C save(C rest) {
+    public C save(C rest, String companyAccountId) throws NoSuchAlgorithmException {
         addEtag(rest);
         addKind(rest);
         addLinks(rest);
         E baseEntity = genericTransformer.transform(rest);
-        addID(baseEntity);
+        baseEntity.setId(generateID(companyAccountId));
```

Comment:
Bare in mind, do we want to keep adding things to the abstract save method when we know it doesn't yet fit in with the company-accounts resource creation? Should we wait until the company-accounts resource stuff has gone in before we keep adding to the abstract service?

Reply:
We should implement what we can at the moment and if the CompanyAccounts doesn't fit the Abstract save we can refactor it out later. I'm concerned if we wait we will end up with a form of analysis paralysis instead we should follow the Agile methodology and incrementally improve what we have.